src/common/IComponent.hpp

```diff
@@ -327,6 +331,8 @@ class IComponent {
     */
    virtual ~IFieldGroup() = default;
 
+   IFieldGroup() = default;
```

Comment:
his seems useless, and even if you want a IFieldGroup constructor it shall be protected and documented.